src/analysis/bounded_variation.lean

```diff
@@ -686,60 +688,133 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
+@[protected]
+lemma nonneg_of_le {a b : α} (h : a ≤ b) : 0 ≤ variation_on_from_to f s a b :=
+by simp only [variation_on_from_to, if_pos h, ennreal.to_real_nonneg]
+
+@[protected]
+lemma eq_neg_swap (a b : α) :
+  variation_on_from_to f s a b = - variation_on_from_to f s b a :=
+begin
+  rcases lt_trichotomy a b with ab|rfl|ba,
+  { simp only [variation_on_from_to, if_pos ab.le, if_neg ab.not_le, neg_neg], },
+  { simp only [self, neg_zero], },
+  { simp only [variation_on_from_to, if_pos ba.le, if_neg ba.not_le, neg_neg], },
+end
+
+@[protected]
+lemma nonpos_of_ge {a b : α} (h : b ≤ a) : variation_on_from_to f s a b ≤ 0 :=
+begin
+  rw eq_neg_swap,
+  exact neg_nonpos_of_nonneg (nonneg_of_le f s h),
+end
+
+@[protected]
+lemma eq_of_le {a b : α} (h : a ≤ b) :
+  variation_on_from_to f s a b = (evariation_on f (s ∩ Icc a b)).to_real := if_pos h
+
+@[protected]
+lemma eq_of_ge {a b : α} (h : b ≤ a) :
+  variation_on_from_to f s a b = - (evariation_on f (s ∩ Icc b a)).to_real :=
+by rw [eq_neg_swap, neg_inj, eq_of_le f s h]
+
+@[protected]
+lemma add {f : α → E} {s : set α} (hf : has_locally_bounded_variation_on f s)
+  {a b c : α} (ha : a ∈ s) (hb : b ∈ s) (hc : c ∈ s) :
+  variation_on_from_to f s a b + variation_on_from_to f s b c = variation_on_from_to f s a c :=
+begin
+  symmetry,
+  refine @additive_of_is_total _ _ _ (≤) _ (λ (x y : s), variation_on_from_to f s x y)
+                               (λ x, true) _ _ ⟨a,ha⟩ ⟨b,hb⟩ ⟨c,hc⟩ trivial trivial trivial,
+  { rintro ⟨x,xs⟩ ⟨y,ys⟩,
+    simp only [eq_neg_swap f s y x, subtype.coe_mk, add_right_neg, forall_true_left], },
+  { rintro ⟨x,xs⟩ ⟨y,ys⟩ ⟨z,zs⟩ xy yz,
+    dsimp only,
+    rw [eq_of_le f s xy,
+        eq_of_le f s yz,
+        eq_of_le f s (xy.trans yz)],
+    dsimp only [variation_on_from_to],
+    rintros _ _ _,
+    rw [←ennreal.to_real_add, evariation_on.Icc_add_Icc f xy yz ys],
+    exacts [hf x y xs ys, hf y z ys zs] }
+end
+
+variables {f} {s}
+
+@[protected]
+lemma monotone_on (hf : has_locally_bounded_variation_on f s)
+  {a : α} (as : a ∈ s) : monotone_on (variation_on_from_to f s a) s :=
+begin
+  rintro b bs c cs bc,
+  rw ←add hf as bs cs,
+  exact le_add_of_nonneg_right (nonneg_of_le f s bc),
+end
+
+@[protected]
+lemma antitone_on (hf : has_locally_bounded_variation_on f s)
+  {b : α} (bs : b ∈ s) : antitone_on (λ a, variation_on_from_to f s a b) s :=
+begin
+  rintro a as c cs ac,
+  dsimp only,
+  rw ←add hf as cs bs,
+  exact le_add_of_nonneg_left (nonneg_of_le f s ac),
+end
+
+@[protected]
+lemma sub_self_monotone_on {f : α → ℝ} {s : set α}
+  (hf : has_locally_bounded_variation_on f s) {a : α} (as : a ∈ s) :
+  monotone_on (variation_on_from_to f s a - f) s :=
+begin
+  rintro b bs c cs bc,
+  rw [pi.sub_apply, pi.sub_apply, le_sub_iff_add_le, add_comm_sub, ← le_sub_iff_add_le'],
+  calc f c - f b
+     ≤ |f c - f b| : le_abs_self _
+  ...= dist (f c) (f b) : real.dist_eq _ _
+  ...= dist (f b) (f c) : dist_comm _ _
+  ...≤ variation_on_from_to f s b c : by
```

Comment:
```suggestion
      ≤ |f c - f b| : le_abs_self _
  ... = dist (f b) (f c) : by rw [dist_comm, real.dist_eq]
  ... ≤ variation_on_from_to f s b c : by
```
You can combine two steps. Also, space before operators.

Reply:
by operators, you mean `abs` in this case?